### ISSUE TITLE
`jsdoc/no-types`: Not reporting violations on TS interfaces

### ISSUE THREAD

**Issue by ej612** [gajus/eslint-plugin-jsdoc#1249]
Hi there!

I have the following eslint config:
```
extends: [
  'plugin:jsdoc/recommended-typescript-error'
]
```
And the following code:
```ts
export class A {
    /**
     * @param {string} paramA
     */
    public methodA(paramA: string) {
        // ...
    }
}
```
`no-types` is warning me not to use types in my JSdoc (as expected):
![image](https://github.com/gajus/eslint-plugin-jsdoc/assets/110408767/1577a121-0bed-4d3f-872b-7435333fe0df)

If however, I don't declare a `class`, but an `interface`, like so:
```ts
export interface B {
    /**
     * @param {string} paramA
     */
    methodB(paramB: string): void
}
```
The rule no longer reports a violation.

Is this expected behavior? I would expect it to report a violation on interfaces as well.

Thanks a lot in advance!

## Environment

- Node version: 18
- ESLint version 8.57.0
- `eslint-plugin-jsdoc` version: 48.2.13

**Comment by brettz9** [comment]
The rules have long been designed this way—I don't know if it was chosen for performance reasons or what, but it would seem to me to make sense to be on by default in more places or as someone requested, to have a config which enables all of the stricter settings (see #701). I'm frankly just not as inclined to work on doing this myself, but a PR would be welcome.

**Comment by brettz9** [comment]
I think we can discuss any further concerns at #701 , so closing for now. Feel free to comment further there as needed.

**Comment by brettz9** [comment]
> Is this really the same issue as #701? It would seem that over there, we're talking about creating one or multiple configs with all rules turned on, whereas here, we're not talking about enabling or disabling rules, but rather broadening the scope on which these rules apply.

Part of #701 would be tweaking the rules with specific options (e.g., enabling more contexts).

> Or are you saying that it wouldn't work to make `no-types` apply to TS interfaces if the environment is plain JS? Wouldn't it simply be ignored? 

Yes, it'd simply be ignored. I think `no-types` is a candidate for just checking all environments, but for some rules, projects might not be interested in enforcing say perfect stylistic structure everywhere.

> If we can't include TS interfaces for all environments, should we change the defaults for the `recommended-typescript-error` config to include interfaces?

I think we can include TS interfaces for all environments.

I'll reopen, as it's fine to just track this for `no-types`, especially as I think it makes sense to apply it everywhere.

**Comment by brettz9** [comment]
I've added handling by default for the case you mentioned. Let me know if others come up.

**Comment by brettz9** [comment]
Dupe of #1225 .
